crates/paths/src/server.rs

```diff
@@ -0,0 +1,176 @@
+use std::fmt::Write;
+use std::{fs, io};
+
+use crate::utils::{path_type, PathBufExt};
+use chrono::NaiveDate;
+
+path_type!(ServerDataDir: dir);
+
+impl ServerDataDir {
+    pub fn config_toml(&self) -> ConfigToml {
+        ConfigToml(self.0.join("config.toml"))
+    }
+
+    pub fn logs(&self) -> LogsDir {
+        LogsDir(self.0.join("logs"))
+    }
+
+    pub fn wasmtime_cache(&self) -> WasmtimeCacheDir {
+        WasmtimeCacheDir(self.0.join("cache/wasmtime"))
+    }
+
+    pub fn metadata_toml(&self) -> MetadataTomlPath {
+        MetadataTomlPath(self.0.join("metadata.toml"))
+    }
+
+    pub fn pid_file(&self) -> Result<PidFile, PidFileError> {
+        use fs2::FileExt;
+        use io::{Read, Write};
+        self.create()?;
+        let path = self.0.join("spacetime.pid");
+        let mut file = fs::File::options()
+            .create(true)
+            .write(true)
+            .truncate(false)
+            .read(true)
+            .open(&path)?;
+        match file.try_lock_exclusive() {
+            Ok(()) => {}
+            Err(e) if e.kind() == io::ErrorKind::WouldBlock => {
+                let mut s = String::new();
+                let pid = file.read_to_string(&mut s).ok().and_then(|_| s.trim().parse().ok());
+                return Err(PidFileError::Exists { pid });
+            }
+            Err(e) => return Err(e.into()),
+        }
+        let mut pidfile = PidFile { file, path };
+        pidfile.file.set_len(0)?;
+        write!(pidfile.file, "{}", std::process::id())?;
+        Ok(pidfile)
+    }
+
+    pub fn replica(&self, replica_id: u64) -> ReplicaDir {
```

Comment:
What's the motivation for having multiple replica directories?

Reply:
A replica is just what spacetime calls a database instance now, I guess - so each replica directory contains the database for a separate module.